### PR TITLE
shared-utilities Add SharedUtilities Folder and Project

### PR DESCRIPTION
**Pull request by cBridges851**
- Put puzzle projects into their own folder
- Add new SharedUtilities folder with a class library and test project inside, ready for when the `TwoDArrayConverter` is needed for day 6.
- Add new methods in `TwoDArrayConverter`
  - As well as handling splitting by character as per before, you can now use it to separate cells with a given separator, e.g. `,` 
  - Tests added for `TwoDArrayConverter`. Generated by GitHub CoPilot.

## Summary by CodeRabbit

* **New Features**
  * Added a shared utilities library and multiple puzzle solution projects to the solution.
  * Grid conversion utility now supports both character and string grids and configurable column separators.

* **Tests**
  * Extensive unit tests added for the grid converter covering separators, trimming, CRLF, uneven rows, empty/null inputs and many edge cases.

* **Chores**
  * Solution and project metadata updated to include new projects; test tooling versions upgraded.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: AdventOfCode2025/SharedUtilities.Test/UsingTheTwoDArrayConverter/WhenAStringIsPassedInWithASeparator.cs [hunk 1]
```diff
@@ -0,0 +1,111 @@
+﻿namespace SharedUtilities.Test.UsingTheTwoDArrayConverter {
+    public class WhenAStringIsPassedInWithASeparator {
+
```
[the remark this answers is not in the document]
Comment (cBridges851) [reply AdventOfCode2025/SharedUtilities.Test/UsingTheTwoDArrayConverter/WhenAStringIsPassedInWithASeparator.cs:3]: This is a .NET 10 project that uses implicit `using`s
